src/replayq.erl

```diff
@@ -24,6 +24,8 @@
 -type ack_ref() :: ?NOTHING_TO_ACK | {segno(), ID :: pos_integer()}.
 -type sizer() :: fun((item()) -> bytes()).
 -type marshaller() :: fun((item()) -> binary()).
+-type accumulator() :: term().
+-type stop_function() :: fun((item(), accumulator()) -> true | accumulator()).
```

Comment:
Nit: seems that the only accumulating option that makes sense is cons (because we explicitly apply reverse to the acc), so why ask to implement it in the callback?
m.b. just 
```
-type need_accumulate() :: fun((item(), accumulator()) -> true | false).
```

Reply:
Not sure if I understand. The function is added because we want a way to get batches in emqx_resource_buffer_worker that only contain a certain kinds of requests. It would be very insufficient to scan all items pop'ed so far every time a new item is added to the list.